src/Dialog.js

```diff
@@ -0,0 +1,162 @@
+// @flow
+import React from "react"
+import { MDCDialog } from "@material/dialog/dist/mdc.dialog"
+
+import Button from "./Button"
+
+import type { ChildrenArray } from "react"
+
+type DialogProps = {
+  id: string,
+  title: string,
+  open: boolean,
+  validateOnClick?: boolean,
+  cancelText?: string,
+  submitText?: string,
+  children: ChildrenArray<any>,
+  onAccept?: () => void,
+  onCancel?: () => void,
+  hideDialog: () => void
+}
+
+export default class Dialog extends React.Component<*, *> {
+  props: DialogProps
+
+  dialog = null
+  dialogRoot = null
+
+  constructor(props: DialogProps) {
+    super(props)
+    // this avoids showing the dialog before the CSS has loaded
+    this.state = {
+      style: {
+        display: "none"
+      }
+    }
+  }
+
+  componentDidMount() {
+    const { open } = this.props
+
+    // Hack to get dialog to play nicely with JS tests
+    if (!this.dialogRoot || !this.dialogRoot.dataset) return
+
+    this.dialog = new MDCDialog(this.dialogRoot)
+    this.attachDialogListeners(this.dialog)
+
+    if (open) {
+      this.showMdc()
+    }
+  }
+
+  componentWillUnmount() {
+    this.destroyMdc()
+  }
+
+  componentWillReceiveProps(nextProps: DialogProps) {
+    if (this.props.open !== nextProps.open) {
+      if (nextProps.open) {
+        this.showMdc()
+      } else {
+        this.closeMdc()
+      }
+    }
+  }
+
+  // This function only exists because of false Flow errors
+  attachDialogListeners = (dialog: Object) => {
+    const { onAccept, onCancel, hideDialog, validateOnClick } = this.props
+
+    if (onAccept) {
+      dialog.listen("MDCDialog:accept", onAccept)
+    }
+    if (!validateOnClick) {
+      dialog.listen("MDCDialog:accept", hideDialog)
+    }
+    if (onCancel) {
+      dialog.listen("MDCDialog:cancel", onCancel)
+    }
+    dialog.listen("MDCDialog:cancel", hideDialog)
+  }
+
+  showMdc() {
+    this.setState(() => ({}))
```

Comment:
is this to trigger a re-render?

Reply:
This avoids having `display: none` applied to the dom after the initial render. That style is only added to avoid a flash of unstyled dialogs.

Reply:
gotcha